src/background/utils/preinject.js

```diff
@@ -282,17 +285,25 @@ function prepareScript(script) {
   const reqsSlices = reqs ? [].concat(...reqs.map(req => [req, '\n;'])) : [];
   const hasReqs = reqsSlices.length;
   const wrap = !meta.unwrap;
+  const { grant } = meta;
+  const numGrants = grant.length;
+  const grantNone = !numGrants || numGrants === 1 && grant[0] === 'none';
   const injectedCode = [
+    wrap && `window.${dataKey}=function(${
+      // using a shadowed name to avoid scope pollution
+      grantNone ? GRANT_NONE_VARS : 'GM'}${
+      IS_FIREFOX ? `,${dataKey}){try{` : '){'}${
+      grantNone ? '' : 'with(this)with(c)delete c,'
     // hiding module interface from @require'd scripts so they don't mistakenly use it
-    wrap && `window.${dataKey}=function(module,${dataKey}){try{with(module)((define,module,exports)=>{`,
+    }((${grantNone ? 'define,module,exports' : ''})=>{`,
```

Comment:
This line is breaking UMD requirements in pages using RequireJS when `@grant` is not none.